src/mixins/java/org/spongepowered/common/mixin/movementcheck/server/network/ServerGamePacketListenerImplMixin_MovementCheck.java

```diff
@@ -40,53 +39,59 @@ public abstract class ServerGamePacketListenerImplMixin_MovementCheck {
+        )
+    )
+    private float movementCheck$onMovedTooQuickly(final float value) {
+        final double threshold = SpongeGameConfigs.getForWorld(this.player.level).get().movementChecks.movedTooQuicklyThreshold;
+        if (threshold > 0.0D) {
+            return (float) threshold;
         }
-        return true; // The 'moved too quickly' check only executes if isChangingDimension return false
+        return Float.NaN;
     }
 
-    @Redirect(method = "handleMovePlayer",
-        at = @At(value = "INVOKE", target = "Lnet/minecraft/server/level/ServerPlayer;isChangingDimension()Z", ordinal = 1))
-    private boolean movementCheck$onMovedWronglyCheck(final ServerPlayer player) {
-        if (SpongeGameConfigs.getForWorld(this.player.level).get().movementChecks.movedWrongly) {
-            return player.isChangingDimension();
+    @ModifyConstant(method = "handleMovePlayer", constant = @Constant(doubleValue = 0.0625D, ordinal = 0), slice =
+        @Slice(
+            from = @At(value = "INVOKE", target = "Lnet/minecraft/server/level/ServerPlayer;isChangingDimension()Z", ordinal = 1),
+            to = @At(value = "INVOKE", target = "Lnet/minecraft/server/level/ServerPlayer;isSleeping()Z", ordinal = 1)
+        )
+    )
+    private double movementCheck$onMovedWronglyCheck(final double value) {
+        final double threshold = SpongeGameConfigs.getForWorld(this.player.level).get().movementChecks.movedWronglyThreshold;
+        if (threshold > 0.0D) {
+            return threshold;
         }
-        return true; // The 'moved too quickly' check only executes if isChangingDimension return false
+        return Double.NaN;
     }
 
-    @ModifyConstant(method = "handleMoveVehicle", constant = @Constant(doubleValue = 100, ordinal = 0), slice =
+    @ModifyConstant(method = "handleMoveVehicle", constant = @Constant(doubleValue = 0.0625D, ordinal = 0), slice =
         @Slice(
-            from = @At(value = "INVOKE", target = "Lnet/minecraft/world/phys/Vec3;lengthSqr()D", ordinal = 0),
-            to = @At(value = "INVOKE", target = "Lnet/minecraft/server/network/ServerGamePacketListenerImpl;isSingleplayerOwner()Z", ordinal = 0))
+            from = @At(value = "INVOKE", target = "Lnet/minecraft/world/entity/Entity;move(Lnet/minecraft/world/entity/MoverType;Lnet/minecraft/world/phys/Vec3;)V", ordinal = 0),
+            to = @At(value = "INVOKE", target = "Lorg/apache/logging/log4j/Logger;warn(Ljava/lang/String;Ljava/lang/Object;Ljava/lang/Object;Ljava/lang/Object;)V")
+        )
     )
-    private double movementCheck$onVehicleMovedTooQuicklyCheck(final double val) {
-        if (SpongeGameConfigs.getForWorld(this.player.level).get().movementChecks.player.vehicleMovedTooQuickly) {
-            return val;
+    private double movementCheck$onVehicleMovedWronglyCheck(final double value) {
+        final double threshold = SpongeGameConfigs.getForWorld(this.player.level).get().movementChecks.movedWronglyThreshold;
```

Comment:
This is using the player's one instead of the vehicle's. You forgot to introduce `vehicle-moved-wrongly-threshold`

Reply:
A previous configuration knob did not exist seperating player and vehicle and Paper/Spigot doesn't have a seperate one for each either I think. If still think this is useful though I can add it.

Reply:
I think it would be useful. It doesn't hide the fact that the same knob is used for both vehicle and players as the other one is separated like that. Also considering players and vehicles have different physics.